chia/cmds/db_upgrade_func.py

```diff
@@ -314,26 +323,39 @@ async def convert_v1_to_v2(in_path: Path, out_path: Path) -> None:
                     commit_in -= 1
                     if commit_in == 0:
                         commit_in = COIN_COMMIT_RATE
-                        await out_db.executemany("INSERT INTO coin_record VALUES(?, ?, ?, ?, ?, ?, ?, ?)", coin_values)
-                        await out_db.commit()
-                        await out_db.execute("begin transaction")
+                        out_db.executemany("INSERT INTO coin_record VALUES(?, ?, ?, ?, ?, ?, ?, ?)", coin_values)
+                        out_db.commit()
+                        out_db.execute("begin transaction")
                         coin_values = []
                         end_time = time()
                         rate = COIN_COMMIT_RATE / (end_time - start_time)
                         start_time = end_time
 
-            await out_db.executemany("INSERT INTO coin_record VALUES(?, ?, ?, ?, ?, ?, ?, ?)", coin_values)
-            await out_db.commit()
+            out_db.executemany("INSERT INTO coin_record VALUES(?, ?, ?, ?, ?, ?, ?, ?)", coin_values)
+            out_db.commit()
             end_time = time()
             print(f"\r      {end_time - coin_start_time:.2f} seconds                             ")
 
             print("[5/5] build indices")
             index_start_time = time()
             print("      block store")
-            await BlockStore.create(DBWrapper(out_db, db_version=2))
+            out_db.execute("CREATE INDEX height on full_blocks(height)")
+            out_db.execute(
+                "CREATE INDEX is_fully_compactified ON"
+                " full_blocks(is_fully_compactified, in_main_chain) WHERE in_main_chain=1"
```

Comment:
Does `in_main_chain` have a limit, e.g. `(peak-5 blocks)`?

The reason I ask is I'm wondering about the possible danger of new code using this index to mean something slightly different than other "in_main_chain_ equivalent checks elsewhere.

Reply:
`in_main_chain` is a field indicating that the block is part of the current longest chain. whenever we reorg, we also update this field to keep a single trail of blocks from genesis to the peak. One invariant is that every height has exactly one block where `in_main_chain = 1`.